File: server/sequelize/models/Resident.js
import DataTypes from "sequelize";

/**
 * Modelo de 'residente' en base de datos
 * @param {*} sequelize 
 */
export const residentModel = (sequelize) => {
  sequelize.define(
    "Resident",
    {
      residentId: {
        type: DataTypes.INTEGER,
        primaryKey: true,
        autoIncrement: true,
        field: "id_residente",
      },
      email: {
        type: DataTypes.STRING,
        allowNull: false,
        unique: true
      },
      password: {
        type: DataTypes.STRING,
        allowNull: false,
      },
      tower: {
        type: DataTypes.INTEGER,
        allowNull: false,
        field: "torre",
      },
      apartment: {
        type: DataTypes.INTEGER,
        allowNull: false,
        field: "apartamento",
      },
      phoneNumber1: {
        type: DataTypes.STRING,
        field: "telefono1",
      },
      phoneNumber2: {
        type: DataTypes.STRING,
        field: "telefono2",
      },
      phoneNumber3: {
        type: DataTypes.STRING,
        field: "telefono3",
      },
      ownerName: {
        type: DataTypes.STRING,
        allowNull: false,
        field: "nombre_propietario",
      },
      parkingActive: {
        type: DataTypes.BOOLEAN,
        allowNull: false,
        defaultValue: false,
        field: "parqueado_activo"
      },
      deviceToken: {
        type: DataTypes.STRING,
        field: "token_dispositivo"
      },
      webPage: {
        type: DataTypes.STRING,
        allowNull: false,
        defaultValue: "",
        field: "pagina_web"
      }
    },
    {
      tableName: "residente",
      timestamps: false,
    }
  );
};

File: mi_conjunto_guardas/lib/models/resident.dart


/// Modelo de 'residente'
class Resident {
  int residentId;
  String email;
  int tower;
  int apartment;
  String phoneNumber1;
  String phoneNumber2;
  String phoneNumber3;
  String ownerName;
  bool parkingActive;

  Resident({
    this.residentId,
    this.email,
    this.tower,
    this.apartment,
    this.phoneNumber1,
    this.phoneNumber2,
    this.phoneNumber3,
    this.ownerName,
    this.parkingActive,
  });

  factory Resident.fromJson(Map<String, dynamic> json) {
    return Resident(
        residentId: json["residentId"],
        email: json["email"],
        tower: json["tower"],
        apartment: json["apartment"],
        phoneNumber1: json["phoneNumber1"],
        phoneNumber2: json["phoneNumber2"],
        phoneNumber3: json["phoneNumber3"],
        ownerName: json["ownerName"],
        parkingActive: json["parkingActive"]);
  }
}
